### PR TITLE
Python comprehensions

### PR DESCRIPTION
**Pull request by yangmillstheory**
Reopening #973 to merge from feature branch. Also fixed a `SnipMateSyntaxError` for `snippets/python.snippets`.

### CODE REVIEW

File: UltiSnips/python.snippets [hunk 1]
```diff
@@ -709,4 +709,16 @@ snippet pmdoc "pocoo style module doc string" b
...
+# Comprehensions #
+##################
+
+snippet lcp "list comprehension"
+[$1 for $2 in ${3:${VISUAL}}]$0
+endsnippet
+
+snippet dcp "dict comprehension"
+{$1: $2 for $3 in ${4:${VISUAL}}}$0
+endsnippet
```
Comment (lpil) [inline UltiSnips/python.snippets]: Remove these please, no reason to duplicate the snippets if they don't use any Ultisnips specific features :)

Comment (yangmillstheory) [reply UltiSnips/python.snippets]: Sorry, I'm not sure what you mean, what should I remove? I'm new to developing against UltiSnips, just a casual user.

Comment (lpil) [reply UltiSnips/python.snippets]: No problem! All plugins can read the snipmate format, so there's no need to duplicate snippets in both formats. The two you've added in the ultisnips format can be safely removed :)

Comment (yangmillstheory) [reply UltiSnips/python.snippets]: Thanks, mind blown 👍 

mindblown

**Pull request merged**